ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTableActions/NetworkTableActions.tsx

```diff
@@ -24,33 +40,36 @@ const NetworkTableActions = ({ nic, systemId }: Props): JSX.Element | null => {
   const canMarkAsConnected = true;
   const canMarkAsDisconnected = true;
   const cannotEditInterface = false;
-  const actions = [
-    ...(canMarkAsConnected && [
-      {
-        children: "Mark as connected",
-      },
-    ]),
-    ...(canMarkAsDisconnected && [
-      {
-        children: "Mark as disconnected",
-      },
-    ]),
-    ...(canAddAliasOrVLAN && [
-      {
-        children: "Add alias or VLAN",
-      },
-    ]),
-    ...(!cannotEditInterface && [
-      {
-        children: `Edit ${getInterfaceTypeText(nic)}`,
-      },
-    ]),
-    ...(canBeRemoved && [
-      {
-        children: `Remove ${getInterfaceTypeText(nic)}...`,
-      },
-    ]),
-  ];
+  let actions: { children: string }[] = [];
+  if (machine && nic) {
+    actions = [
+      ...(canMarkAsConnected && [
+        {
+          children: "Mark as connected",
+        },
+      ]),
+      ...(canMarkAsDisconnected && [
+        {
+          children: "Mark as disconnected",
+        },
+      ]),
+      ...(canAddAliasOrVLAN && [
+        {
+          children: "Add alias or VLAN",
+        },
+      ]),
+      ...(!cannotEditInterface && [
```

Comment:
Not that it matters for this PR but it seems a bit odd that the rest of the actions are `can` and this one is `!cannot`

Reply:
Hmm yeah good point, that was probably me following the legacy utils. I'll see if the logic can be reversed when I implement the util.